scripts/check_doc_links.py

```diff
@@ -46,7 +53,12 @@ def is_url_reachable(url: str) -> bool:
         return True
     try:
         response = requests.head(url)
-        return response.status_code in [200, 403, 405]
+        if response.status_code == 200:
+            return True
+        elif response.status_code in [403, 405]:
+            return WHITELIST_URL_TO_CODE.get(url, 404) in [403, 405]
+        else:
+            return False
```

Comment:
that is OK; however, to be more general, you could make the whitelist to be valid for any code.

That is:
```python
return url in whitelist or requests.head(url).status_code == 200
```
plus, `WHITELIST_URL_TO_CODE` as a collection (e.g. list or set)